### DISCUSSION TITLE
Accesible Progress Bar

### DISCUSSION THREAD

**Discussion by HectorSancho** [Telefonica/mistica-design#1651] [Proposals]
### Goal

Resolve the progress bar accesibility issues

### Context

We have two accesibility problems with the current definition of the progress bar:

- Contrats between control activated and control in some brands
- Contrast between the token control and a white background (don't pass double AA contrast)

### Impact

Variable, in some proposals the changes would affect the component's structure. Other options only require change of tokens

### Discussion Questions

- Should color of tokens be 100% accesible? or are there other ways to ensure acccesibility depending of the context? (ejem helper texts)
- Should this component pass accessibility tests or is it a complementary element to a data or visual support in other cases?

### Do you have a visual proposal?

https://www.figma.com/file/3ovTLfO36BHwD1NxoBaHES/Progress-Bar-accesibility-explorations?type=design&node-id=4%3A17&mode=design&t=lrAUYyOWWTDiuCPu-1

**Comment by HectorSancho** [comment]
Screenshot:

<img width="610" alt="Captura de pantalla 2024-02-14 a las 13 30 30" src="https://github.com/Telefonica/mistica-design/assets/56122343/b9b77fde-a208-480d-82bb-e42d8b3a0866">

Material Design reference: https://m3.material.io/components/progress-indicators/accessibility

**Comment by aweell** [comment]
Another topic that we need to consider about the progress bars is avoid failing:

[SC 1.4.1. Use of Color (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color)

Now we delegate this responsibility in the correct usage of the component in a context that displays a textual information related to the value of the progress bar and also min and max values.

Should be something we also provide by default with the possibility of overrides?

**Reply by aweell** [reply]
The options that I see (to avoid failing 1.4.1)
 
* These blocks are already predefined and we pass the label internally like the live blocks
* We put the label on it, conditioning the use cases and that that has to be used.
* We give you a prop label that can come empty, if you fill it out you paint the default label, if you want another one you pass it through labeledby as before
* It stays as it is, which is that you paint the label however you want and use labelledby

The last two points delegate responsibility to the teams.

**Comment by aweell** [comment]
I add here the problems that are added when adapting the progress to work in inverse contexts.

Here a solution similar to A.1 proposal

![Screenshot 2024-10-07 at 09 48 58](https://github.com/user-attachments/assets/c5a350a8-dbdd-4469-bf78-1ca7e5cedcef)

**Comment by aweell** [comment]
Will be solved in: https://github.com/Telefonica/mistica-design/issues/2389

I close this discussion
